File: Testing/Stdlib Test Code/test_result_bytecode.js

function divide(a, b) {
    if ((b === 0)) {
        return {ok: false, error: "division by zero"};
    }
    return {ok: true, value: Math.trunc(a / b)};
}

function main() {
    const _r0 = divide(10, 2);
    if (!_r0.ok) {
        console.error("error: " + _r0.error);
        process.exit(1);
    }
    let r1 = _r0.value;
    console.log(r1);
    const _r1 = divide(100, 5);
    if (!_r1.ok) {
        console.error("error: " + _r1.error);
        process.exit(1);
    }
    let r2 = _r1.value;
    console.log(r2);
    const _r2 = divide(42, 7);
    if (!_r2.ok) {
        console.error("error: " + _r2.error);
        process.exit(1);
    }
    let r3 = _r2.value;
    console.log(r3);
    return 0;
}

function __ul_sort(arr, n) { const sub = arr.slice(0, n).sort((a, b) => a < b ? -1 : a > b ? 1 : 0); for (let i = 0; i < n; i++) arr[i] = sub[i]; }

main();

File: Testing/Stdlib Test Code/test_result_bytecode.py
from dataclasses import dataclass, field
import sys

def __ul_sort(arr, n): arr[:n] = sorted(arr[:n])

class __ul_Exception(Exception):
    def __init__(self, type_name, message):
        self.type_name = type_name
        self.message = message
        super().__init__(message)


def divide(a, b):
    if (b == 0):
        return (False, 'division by zero')
    return (True, (a // b))

def main():
    _r0 = divide(10, 2)
    if not _r0[0]:
        print("error: " + str(_r0[1]), file=sys.stderr)
        sys.exit(1)
    r1 = _r0[1]
    print(r1)
    _r1 = divide(100, 5)
    if not _r1[0]:
        print("error: " + str(_r1[1]), file=sys.stderr)
        sys.exit(1)
    r2 = _r1[1]
    print(r2)
    _r2 = divide(42, 7)
    if not _r2[0]:
        print("error: " + str(_r2[1]), file=sys.stderr)
        sys.exit(1)
    r3 = _r2[1]
    print(r3)
    return 0


if __name__ == "__main__":
    main()
